main.go

```diff
@@ -104,6 +106,31 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "HostDeviceNetwork")
 		os.Exit(1)
 	}
+
+	upgradeLogger := ctrl.Log.WithName("controllers").WithName("Upgrade")
+	k8sInterface, err := utils.CreateK8sInterface()
+	if err != nil {
+		setupLog.Error(err, "unable to create k8s interface", "controller", "Upgrade")
+		os.Exit(1)
+	}
+	nodeUpgradeStateProvider := upgrade.NewNodeUpgradeStateProvider(
+		mgr.GetClient(), upgradeLogger.WithName("nodeUpgradeStateProvider"))
+	drainManager := upgrade.NewDrainManager(
+		k8sInterface, nodeUpgradeStateProvider, upgradeLogger.WithName("drainManager"))
+	podRestartManager := upgrade.NewPodRestartManager(mgr.GetClient(), upgradeLogger.WithName("podRestartManager"))
+	clusterUpdateStateManager := upgrade.NewClusterUpdateStateManager(
```

Comment:
As far as I can see you don't use k8sInterface in ClusterUpdateStateManager. Please, remove if it is not required.

Reply:
It's used for node uncordon

Reply:
ack